### PR TITLE
Introducing Excel look and feel and column specified table filtering

### PR DESCRIPTION
**Pull request by Haripriya-Daggubati**
Updated table filtering in such a way that the user feels the Excel type filtering.

In previous scenarios of table filtering, we needed to right-click on the column header to view the search pop-up window. Now we added an icon in the column header for filtering options, and by just clicking on the icon, it shows the search pop-up window.

Excel_filtering

And introducing specific column searching along with table searching (search option available for all columns in the table).

Specific_column_option

To differentiate filtered columns as part of Excel filtering, introduced two different icons for before and after filtering and also provided the privilege to the user to change the icons as per their needs.

Before_and_after_filtering_icon

Not as a disadvantage, to more like excel feel removed clear table option in search pop-up window.

### CODE REVIEW

File: maven/repository/org/oxbow/swing-bits/1.0.0-SNAPSHOT/maven-metadata-local.xml [hunk 1]
```diff
@@ -1,35 +1,35 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <metadata modelVersion="1.1.0">
-  <groupId>org.oxbow</groupId>
-  <artifactId>swing-bits</artifactId>
-  <version>1.0.0-SNAPSHOT</version>
+  <groupId>org.oxbow.se</groupId>
```
Comment (eugener) [inline maven/repository/org/oxbow/swing-bits/1.0.0-SNAPSHOT/maven-metadata-local.xml:3]: Group and artifact name should not change

File: maven/repository/org/oxbow/swing-bits/1.0.0-SNAPSHOT/swing-bits-1.0.0-SNAPSHOT.pom [hunk 2]
```diff
@@ -1,9 +1,9 @@
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
 	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
 	<modelVersion>4.0.0</modelVersion>
-	<groupId>org.oxbow</groupId>
-	<artifactId>swing-bits</artifactId>
-	<version>1.0.0-SNAPSHOT</version>
+	<groupId>org.oxbow.se</groupId>
```
Comment (eugener) [inline maven/repository/org/oxbow/swing-bits/1.0.0-SNAPSHOT/swing-bits-1.0.0-SNAPSHOT.pom:4]: Group and artifact name should not change

File: maven/repository/org/oxbow/swing-bits/1.0.0-SNAPSHOT/swing-bits-1.0.0-SNAPSHOT.pom [hunk 3]
```diff
@@ -23,7 +23,7 @@
 	<repositories>
 		<repository>
 			<id>oxbow-repository</id>
-			<url>https://github.com/eugener/oxbow/raw/master/maven/repository</url>
```
Comment (eugener) [inline maven/repository/org/oxbow/swing-bits/1.0.0-SNAPSHOT/swing-bits-1.0.0-SNAPSHOT.pom:26]: Please keep the ref to the original repo

File: maven/repository/org/oxbow/swing-bits/maven-metadata-local.xml [hunk 4]
```diff
@@ -1,13 +1,13 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <metadata>
-  <groupId>org.oxbow</groupId>
-  <artifactId>swing-bits</artifactId>
+  <groupId>org.oxbow.se</groupId>
```
Comment (eugener) [inline maven/repository/org/oxbow/swing-bits/maven-metadata-local.xml:3]: group and artifact name should not change

File: swingbits/src/main/java/org/oxbow/swingbits/table/filter/TableFilterColumnPopup.java [hunk 5]
```diff
@@ -170,14 +162,15 @@ protected JComponent buildContent() {
...
             toolbar.setOpaque(false);
-            toolbar.add( new PopupWindow.CommandAction(
-                    bundle.getString( "Clear_ALL_COLUMN_FILTERS" ),
-                    new ImageIcon(getClass().getResource("funnel_delete.png"))) {
-                @Override
-                protected boolean perform() {
-                    return clearAllFilters();
-                }
-            });
+            //removed clear filter for JTable
```
Comment (eugener) [inline swingbits/src/main/java/org/oxbow/swingbits/table/filter/TableFilterColumnPopup.java:165]: In my opinion, we should not just remove it, but have API to enable/disable it with default being disabled

File: swingbits/src/main/java/org/oxbow/swingbits/table/filter/TableRowFilterSupport.java [hunk 6]
```diff
@@ -190,13 +204,39 @@ private void setupHeaderRenderers( TableModel newModel, boolean fullSetup ) {
...
+        if (filteringIcon == null) {
+            filteringIcon = new ImageIcon(getClass().getResource("filtering.png"));
+        }
+        if (filteredIcon == null) {
+            filteredIcon = new ImageIcon(getClass().getResource("filtered.png"));
+        }
+
+        TableCellRenderer headerRenderer = null;
+        switch (filterType) {
+            case DEFAULT:
```
Comment (eugener) [inline swingbits/src/main/java/org/oxbow/swingbits/table/filter/TableRowFilterSupport.java:219]: This is nice - users should be able to choose the renderer

**Review by eugener — CHANGES_REQUESTED** [review]

**Comment by eugener** [comment]
Thank you for a very nice PR!
I had few comments... Once all the issues are resolved, your code can be integrated.

**Review by eugener — APPROVED** [review]
Thank you very much!

**Comment by Haripriya-Daggubati** [comment]
Thank you for accepting changes to merge.

I have a request to do from your side. Can we deploy the latest changes to maven repository. So, everyone directly use the jar.

**Comment by eugener** [comment]
I will try to make it happen within the next week.

**Comment by eugener** [comment]
@Haripriya-Daggubati  

The `v1.3.0` release is available on Maven Central  now 
https://oss.sonatype.org/service/local/repositories/releases/content/org/oxbow/swingbits/

Please let me know if you have any issues.

**Comment by Haripriya-Daggubati** [comment]
Thank you so much!

I have no issues.

**Pull request merged**